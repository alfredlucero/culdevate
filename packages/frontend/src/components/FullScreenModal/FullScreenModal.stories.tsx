import React from "react";
import { storiesOf } from "@storybook/react";
import FullScreenModal from "./index";
import Text from "../Text";
import Heading from "../Heading";
import Button from "../Button";

const onHide = () => {
  console.log("Hide Full Screen Modal!");
};

storiesOf("Common/Full Screen Modal", module).add("Showing Example", () => (
  <FullScreenModal isShowing={true} onHide={onHide}>
    <Heading variant="h3">Full Screen Modal Title</Heading>
    <Text variant="p" className="py-4">
      Full Screen Description
    </Text>
    <div className="flex justify-end pt-4">
      <Button type="button" variant="secondary" onClick={onHide} className="mr-2">
        Close
      </Button>
      <Button
        type="button"
        variant="primary"
        onClick={() => {
          console.log("Action!");
        }}
      >
        Action
      </Button>
    </div>
  </FullScreenModal>
));
