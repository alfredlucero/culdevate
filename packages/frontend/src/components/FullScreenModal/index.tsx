import React from "react";
import cn from "classnames";
import ReactDOM from "react-dom";
import { CommonProps } from "../commonProps";

export interface FullScreenModalProps extends CommonProps {
  isShowing: boolean;
  onHide: () => void;
  children: React.ReactNode;
}

const FullScreenModal: React.FC<FullScreenModalProps> = ({
  isShowing,
  onHide,
  children,
  testId = "",
  className = "",
  ...passThroughProps
}) => {
  if (!isShowing) {
    return null;
  }

  return ReactDOM.createPortal(
    <div
      className={cn("modal", "fixed", "w-full", "h-full", "top-0", "left-0", className)}
      {...(testId !== "" ? { "data-testid": testId } : {})}
      {...passThroughProps}
    >
      <div
        className="modal-overlay absolute w-full h-full bg-gray-100 opacity-75"
        data-testid="fullScreenModalOverlay"
      />
      <div
        className="modal-container fixed w-full h-full z-50 overflow-y-auto"
        aria-modal
        aria-hidden
        tabIndex={-1}
        role="dialog"
      >
        <div className="modal-content container mx-auto h-auto text-left py-6">
          <div className="modal-header w-full flex justify-end">
            <svg
              className="fill-current text-gray-700 h-10 w-10"
              role="button"
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 20 20"
              data-dismiss="modal"
              aria-label="Close"
              onClick={onHide}
              data-testid="fullScreenModalCloseButton"
            >
              <title>Close</title>
              <path d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z" />
            </svg>
          </div>
          {children}
        </div>
      </div>
    </div>,
    document.body,
  );
};

export default FullScreenModal;
