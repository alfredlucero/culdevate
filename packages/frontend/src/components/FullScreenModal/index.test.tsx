import React from "react";
import { render, fireEvent } from "@testing-library/react";
import FullScreenModal from "./index";

describe("<FullScreenModal />", () => {
  const fullScreenModalTestId = "fullScreenModalTestId";

  test("should render without error", () => {
    const { container } = render(
      <FullScreenModal isShowing={true} onHide={() => {}} testId={fullScreenModalTestId} className="center-modal">
        Modal
      </FullScreenModal>,
    );

    expect(container).toMatchSnapshot();
  });

  test("should not exist when the full screen modal is not showing", () => {
    const { queryByTestId } = render(
      <FullScreenModal isShowing={false} onHide={() => {}} testId={fullScreenModalTestId}>
        Modal
      </FullScreenModal>,
    );

    expect(queryByTestId(fullScreenModalTestId)).toBeNull();
  });

  test("should call onHide callback when clicking the close icon button", () => {
    const onHideMock = jest.fn();
    const { getByTestId } = render(
      <FullScreenModal isShowing={true} onHide={onHideMock} testId={fullScreenModalTestId}>
        Modal
      </FullScreenModal>,
    );

    fireEvent.click(getByTestId("fullScreenModalCloseButton"));

    expect(onHideMock).toHaveBeenCalled();
  });
});
