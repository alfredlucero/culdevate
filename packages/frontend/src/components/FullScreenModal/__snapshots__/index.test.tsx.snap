// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullScreenModal /> should render without error 1`] = `<div />`;
